assets/src/design-system/components/textArea/index.js

```diff
@@ -0,0 +1,206 @@
+    resize: none;
+
+    ${themeHelpers.expandPresetStyles({
+      preset: {
+        ...theme.typography.presets.paragraph[
+          THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL
+        ],
+      },
+      theme,
+    })};
+
+    :disabled {
+      color: ${theme.colors.fg.disable};
+      border-color: ${theme.colors.border.disable};
+    }
+
+    :active:enabled {
+      color: ${theme.colors.fg.primary};
+    }
+  `
+);
+
+const Counter = styled.div`
+  text-align: right;
+  align-self: flex-end;
+  span {
+    color: ${({ theme }) => theme.colors.fg.tertiary};
+  }
+`;
+
+export const TextArea = forwardRef(
+  (
+    {
+      className,
+      disabled,
+      hasError,
+      hint,
+      id,
+      label,
+      onBlur,
+      onFocus,
+      value,
+      showCount = false,
+      maxLength,
+      isIndeterminate = false,
+      ...props
+    },
+    ref
+  ) => {
+    const textAreaId = useMemo(() => id || uuidv4(), [id]);
+    const textAreaRef = useRef(null);
+    const [focused, setFocused] = useState(false);
+
+    const hasMaxLength = typeof maxLength === 'number';
+    const hasCounter = showCount && hasMaxLength;
+
+    const { handleBlur, handleFocus } = useInputEventHandlers({
+      forwardedRef: ref,
+      inputRef: textAreaRef,
+      focused,
+      setFocused,
+      onBlur,
+      onFocus,
+    });
+
+    let displayedValue = hasMaxLength
+      ? value?.substring(0, maxLength - 1)
+      : value;
+    if (isIndeterminate) {
+      // Display placeholder if value couldn't be determined.
+      displayedValue = '';
+    }
+
+    return (
+      <Container className={className}>
+        {label && (
+          <Label htmlFor={textAreaId} forwardedAs="label" disabled={disabled}>
+            {label}
+          </Label>
+        )}
+        <InputContainer focused={focused} hasError={hasError}>
+          <StyledTextArea
+            id={textAreaId}
+            disabled={disabled}
+            ref={ref || textAreaRef}
+            onBlur={handleBlur}
+            onFocus={handleFocus}
+            value={displayedValue}
+            {...props}
+          />
+          {hasCounter && (
+            <Counter>
+              <Text
+                as="span"
+                size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.X_SMALL}
+              >{`${value.length}/${maxLength}`}</Text>
+            </Counter>
+          )}
+        </InputContainer>
+        {hint && <Hint hasError={hasError}>{hint}</Hint>}
+      </Container>
+    );
+  }
+);
+
+const TextAreaPropTypes = {
+  'aria-label': labelAccessibilityValidator,
+  className: PropTypes.string,
+  disabled: PropTypes.bool,
+  hasError: PropTypes.bool,
+  hint: PropTypes.string,
+  id: PropTypes.string,
+  label: labelAccessibilityValidator,
+  onBlur: PropTypes.func,
+  onChange: PropTypes.func.isRequired,
+  onFocus: PropTypes.func,
+  value: PropTypes.string.isRequired,
+  showCount: PropTypes.bool,
+  maxLength: PropTypes.number,
```

Comment:
💭 maybe we could use the textarea's `maxlength` property here instead of `maxLength`. That way we could keep the api consistent.

Passing it directly into textarea should give us some nice accessibility bonuses too for screen readers 🥳